Gradle 9.x.x compatibility

Current `build.gradle` has a deprecation warning about Gradle 9.x.x compatibility which prevents building with newer version of Gradle:

```bash
$ ./gradlew help --warning-mode all

> Configure project :
Build file '/home/liontiger/work/solus/tmp/ipscan/build.gradle': line 32
The org.gradle.api.plugins.JavaPluginConvention type has been deprecated. This is scheduled to be removed in Gradle 9.0. Consult the upgrading guide for further information: https://docs.gradle.org/8.14.3/userguide/upgrading_version_8.html#java_convention_deprecation
        at build_erv1k1y8g1n08951pqijjztwc.run(/home/liontiger/work/solus/tmp/ipscan/build.gradle:32)
        (Run with --stacktrace to get the full stack trace of this deprecation warning.)
```

With Gradle 9.5.0 it fails like this:

```bash
$ gradle help

FAILURE: Build failed with an exception.

* Where:
Build file '/home/liontiger/work/solus/tmp/ipscan/build.gradle' line: 32

* What went wrong:
A problem occurred evaluating root project 'ipscan'.
> Could not set unknown property 'targetCompatibility' for root project 'ipscan' of type org.gradle.api.Project.

* Try:
> Run with --stacktrace option to get the stack trace.
> Run with --info or --debug option to get more log output.
> Run with --scan to get full insights from a Build Scan (powered by Develocity).
> Get more help at https://help.gradle.org.

BUILD FAILED in 683ms
```

--- a/build.gradle
+++ b/build.gradle
@@ -29,7 +29,7 @@ String exec(def line, String dir = ".", boolean failOnError = true) {
 
 version = exec("git describe --tags", ".", false) ?: "unknown"
 
-sourceCompatibility = targetCompatibility = 17
+java.sourceCompatibility = java.targetCompatibility = 17
 
 def javaModules = ["java.base", "java.prefs", "java.logging", "jdk.crypto.ec"]
 project.ext.set("javaModules", javaModules)

